### DISCUSSION TITLE
frontend translation

### DISCUSSION THREAD

**Discussion by aminesig** [h2oai/wave#2106] [Q&A]
Is there a way to translate the widgets from english to other languages for the frontend visualisation?

**Comment by marek-mihok** [comment]
You can use [python-i18n](https://pypi.org/project/python-i18n/) for translations. Please let me know if this is what you are looking for.

**Reply by aminesig** [reply]
I can't find any docs on how to translate the final result of wave h2o in the browser, take for example the date and time picker, the days and months are in english, how do I translate them to French ? 
if python-i18n can help me, I can't find any attributes to access in wave widgets so I can translate their rendered content in the browser...! 
![image](https://github.com/h2oai/wave/assets/2048486/9c6a89a4-5059-45cb-9faf-81e5208969b5)

**Comment by mturoci** [comment]
Most of the Wave widgets support supplying arbitrary text via labels etc. However, a few components here and there might have hardcoded strings as you correctly pointed out, e.g., date picker. You can file a feature request to allow additional tweaks for the ones you need.

**Reply by aminesig** [reply]
Will do! thanks
